### PR TITLE
Book suggestions test

### PR DESCRIPTION
**Pull request by facucompe**
# Sumary
Add tests for book suggestions route
# Trello Card
https://trello.com/c/aZI4sH4N/31-crear-test-para-endpoint-de-creaci%C3%B3n-de-un-booksuggestion

### CODE REVIEW

File: spec/api/v1/book_suggestions_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      it 'creates a new book suggestion' do
+        expect do
+          post :create, params: { book_suggestion: new_book_suggestion }
+        end.to change { BookSuggestion.count }.by(1)
+      end
+
+      it 'responds with 201 status' do
+        post :create, params: { book_suggestion: new_book_suggestion }
+        expect(response).to have_http_status(:created)
+        expect(JSON.parse(response.body)['user_id']).to eq user.id
```
Comment (alebian) [inline spec/api/v1/book_suggestions_controller_spec.rb]: use response_body helper

File: app/serializers/book_suggestion_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class BookSuggestionSerializer < ActiveModel::Serializer
+  attributes :editorial, :price, :title, :author, :publisher, :year, :link, :user_id
```
Comment (alebian) [inline app/serializers/book_suggestion_serializer.rb]: Do you have to send user_id or the user info? What does Apiray say?
If you need the info, yo can write belongs_to :user

File: .env [hunk 3]
```diff
@@ -0,0 +1,5 @@
+MAILER_DOMAIN=smtp.mailtrap.io
```
Comment (ziflit) [inline .env]: So, you should never commit `.env` files. Usually these files have sensitive information like access keys (like here, username and password). What we do instead usually is
1. Add `.env` to `.gitignore` like you did below
2. Add a `.env-example` or something of the like. This file has the name of all the envvars one should have configured to make the project work. Yours would look like
```
MAILER_DOMAIN=some_domain
MAILER_USERNAME=some_username
MAILER_PASSWORD=my_secure_password
MAILER_PORT=some_port
MAILER_ADDRESS=mailer_address
```
and then anyone who comes later to the project knows which envvars they need.
3. Share our secret `.env` via some secure chat, like Slack or Email.

File: .gitignore [hunk 4]
```diff
@@ -33,7 +33,4 @@
 .elasticbeanstalk/*
 !.elasticbeanstalk/*.cfg.yml
 !.elasticbeanstalk/*.global.yml
-
 .env
```
Comment (ziflit) [inline .gitignore:36]: .env file is part of `.gitignore`. I guess if you just `git rm --cached .env` the file will still be on your PC locally but won't be tracked by git.

File: config/locales/en.yml [hunk 5]
```diff
@@ -1,33 +1,12 @@
...
-#
-# en:
-#   'true': 'foo'
-#
-# To learn more, please read the Rails Internationalization guide
-# available at http://guides.rubyonrails.org/i18n.html.
-
 en:
-  hello: "Hello world"
+  mail_notification_subject: 'You just rent a book!!.'
```
Comment (ziflit) [inline config/locales/en.yml]: So usually what we do in things like this is having the `yml` file scoped with each underlying entity. For example you can do 
```
notification_mail:
  title: 'Book rental detail'
  subject: 'You just rent a book!'
book:
  title: 'Title'
  publisher: 'Publised by'
  ...
```
and then you can do `I18n.t('book.title')` to look for that translation. Read more here https://guides.rubyonrails.org/i18n.html#looking-up-translations

File: app/mailers/notification_generator_mailer.rb [hunk 6]
```diff
@@ -1,6 +1,7 @@
 class NotificationGeneratorMailer < ApplicationMailer
   def new_rent_notification
     @rent = params[:rent]
-    mail to: @rent.user.email, subject: 'Acabas de alquilar un libro!!.'
+    I18n.locale = @rent.user.locale
+    mail to: @rent.user.email, subject: t('mail_notification_subject')
```
Comment (ziflit) [inline app/mailers/notification_generator_mailer.rb]: Read the comment on config/locales/en.yml

File: app/controllers/application_controller.rb [hunk 7]
```diff
@@ -1,3 +1,8 @@
 class ApplicationController < ActionController::Base
   include Wor::Paginate
+  rescue_from ActiveRecord::RecordInvalid, with: :render_validation_error
+
+  def render_validation_error
+    render json: { message: 'Validation failed' }, status: :unprocessable_entity
```
Comment (ziflit) [inline app/controllers/application_controller.rb]: I'm not sure about this approach. What this do is inform there was an error but it doesn't specify where, so the client using the API doesn't know where to fix it.

Usually we would do something like (in the controller)

```ruby
rent = Rent.new(create_params)
if rent.save
  # perform actions on success
else
  render json: { errors: rent.errors }, status: :bad_request
```
`save` returns true/false on success/error so we don't need all the `rescue_from ActiveRecord::RecordInvalid` in the `ApplicationController`

**Pull request merged**